server-utils/src/stream_codec.rs

```diff
@@ -1,8 +1,56 @@
-use std::io;
-use server_utils::tokio_io::codec::{Decoder, Encoder};
+use std::{io, str};
+use tokio_io::codec::{Decoder, Encoder};
 use bytes::BytesMut;
 
-pub struct StreamCodec;
+/// Separator for enveloping messages in streaming codecs
+pub struct Separator(Option<u8>);
```

Comment:
so i could make a separate `impl Default` for it
along with other custom methods